packages/suite/src/support/suite/ConditionalRender.tsx

```diff
@@ -0,0 +1,41 @@
+import { useResponsiveContext } from './ResponsiveContext';
+
+export type ContainerType = 'content' | 'sidebar';
+
+export type UseConditionalRenderProps = {
+    container: ContainerType;
+    minWidth?: number;
+    maxWidth?: number;
+    width?: number;
+};
+
+export type ConditionalRenderProps = UseConditionalRenderProps & {
+    children: React.ReactNode;
+};
+
+const map = {
+    content: 'contentWidth',
+    sidebar: 'sidebarWidth',
+} as const satisfies Record<ContainerType, string>;
+
+export const useConditionalRender = ({
+    container,
+    minWidth,
+    maxWidth,
+    width,
+}: UseConditionalRenderProps) => {
+    const responsiveContext = useResponsiveContext();
+    const containerSize = responsiveContext[map[container]];
+
+    const satisfiesMinWidth = containerSize && minWidth ? containerSize > minWidth : true;
+    const satisfiesMaxWidth = containerSize && maxWidth ? containerSize <= maxWidth : true;
+    const satisfiesWidth = containerSize && width ? containerSize === width : true;
+
+    return satisfiesMinWidth && satisfiesMaxWidth && satisfiesWidth;
+};
+
+export const ConditionalRender = ({ children, ...rest }: ConditionalRenderProps) => {
```

Comment:
Is there any chance for the props to have other values but minWidthMaxWIddth, etc?

I mean, it could be given as boolean props
```ts
<ConditionalRender minWidthSM>

<ConditionalRender maxWidthXL>
```

But that goes against Adam's idea about "compatible API".
But you wouldn't need to import anything.

Reply:
but Adam has a strong opionion.

Reply:
Agree with both of you. I think we should rethink the whole schema how we use these values. Makes sense not importing them and at the same time we should keep it consistent.